### ISSUE TITLE
The No Playables Assumption

### ISSUE THREAD

**Issue by Zamiell** [hanabi/hanabi.github.io#621]
This is a mix between a "new" and an "old" convention.
To be clear, this convention is how we have historically played.
However, this convention has:
1) not been transcribed into the document
2) not been properly communicated to new members of the group

As a senior member of the group (and the maintainer of the doc), the blame for this rests squarely on me.

<br />

## The No Playables Assumption

- First, see the section on the [Known Bluff](level_12.md#the-known-bluff).
- In our group, it is possible to give a *Bluff*. It is also possible to give a *Known Bluff*, but those are much more rare. (*Known Bluffs* comprise around 5% or less of all *Bluffs*.)
- For example, in a 3-player game (a normal *Bluff*):
  - Red 2 is played on the stacks. All of the other 1's are played on the stacks.
  - Cathy's hand is as follows, from left-to-right: `red 4, red 5, blue 1, blue 1, green 1`
  - Alice clues red to Cathy, touching the red 4 on slot 1 and the red 5 on slot 2.
  - Bob tries to play the red 3 into the *Finesse*. Instead, he plays a blue 2, and it successfully plays on the stacks.
  - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a *Bluff*. Since slot 1 was the focus of the clue, it must be the *one-away-from-playable* red card.)
  - What note should Cathy write on her slot 2 card? Cathy starts to write a note of [red 5], but then she stops and considers an alternate hypothetical.
- For example, in a 3-player game (a *Known Bluff*):
  - Red 2 is played on the stacks. All of the other 1's are played on the stacks.
  - Cathy's hand is as follows, from left-to-right: `red 4, red 3, blue 1, blue 1, green 1`
  - Alice clues red to Cathy, touching the red 4 on slot 1 and the red 3 on slot 2.
  - Bob sees that this is a weird clue, because Alice has a *Free Choice* between cluing red and cluing number 3. Bob sees that Alice should have clued number 3. Instead, Alice is intentionally focusing the wrong card.
  - Bob blind-plays his *Finesse Position* as a *Known Bluff*. It is a blue 2 and it successfully plays on the stacks.
  - Cathy marks her slot 1 card as the red 4. (Cathy knows that Alice performed a *Bluff*. Since slot 1 was the focus of the clue, it must be the *one-away-from-playable* red card.)
  - From Cathy's perspective, this new situation is indistinguishable from the situation in the first example. And if Cathy wrote a note of [red 5], in this hypothetical, it would be a mistake!
  - However, Cathy also knows that *Known Bluffs* using *Free Choice* are extremely rare. As an unstated convention that is not officially in the document (but learned through experience in playing with the group), Cathy knows that she should disregard this rare possibility, and instead just simply write a note of [red 5].
  - If Alice *really did* perform a *Known Bluff* with *Free Choice*, then she *Lied* to Cathy about the identity of the slot 2 card. In that case, the team must give a follow-up *Fix Clue* to Cathy at some point in the future.

I acknowledge that Cathy's ignoring of the *Known Bluff* case here may be surprising to some of our newer players, and maybe some older players too. If it is surprising to you, then it is a failure on my part to communicate this properly when I taught you how to play and/or taught you our conventions. (Or, alternatively, if I did not teach you how to play, then a failure on my part for communicating it properly to whoever taught you.)

Cathy's behavior here is called the *No Playables Assumption*. I propose that it is made "official" and that it gets a section in the document. I propose that the text would be something like the following:

- Normally, players who receive a clue after a *Bluff* happens write notes on their cards with their possibilities.
- By default, the players writing the notes should not assume that the *Bluff* was a *Known Bluff* initiated with *Free Choice*, since that move is extremely rare.
- Additionally, players writing the notes should not assume that the *Bluff* was tricking Bob to play into a *Duplication Finesse*, since that move is extremely rare.
- In other words, players should always assume that none of the touched cards in their hand that triggered a *Bluff* are playable - the *No Playables Assumption*.
- The *No Playables Assumption* allows players to narrow the possibilities of what cards in their hand should be. (And since *Known Bluffs* with *Free Choice* are extremely rare, it does not "cost" much to do so.)
- Note that the *No Playables Assumption* does **not** prescribe behavior for Bob - it **only** prescribes behavior for Cathy. Bob should always blind-play his *Finesse Position* card as a *Finesse* or a *Known Bluff* if he sees that Alice clearly does not intend for him to give a OOO *Fix Clue*.
- For example, in a 3-player game:
  - [above example copy-pasted here]

BULLET POINT 6 ABOVE IS IMPORTANT, PLEASE READ IT.

Now, here's some reasons *why* you should not be opposed to this.

<br />

## Interaction With Superpositions (advantage)

- The first obvious advantage of the of having NPA is that it allows players to reduce the possibilities in a superposition.
- This is fairly straightforward. The less possibilities, the less writing notes that people have to do. The less writing notes that people have to do, the less complexity. The less complexity, the less opportunity for mistakes.
- Furthermore, when a player has more information about their hand, it empowers them to give better clues. In the Hyphen-ated group, we we want to empower our teammates with *more* information, not less. This allows expert players to best take advantage of any particular game state.

<br />

## Interaction With Prompts (advantage)

- Another primary advantage of NPA is that it prevents a lot of *Wrong Prompts*.
- Building on the previous example, think about if later on in the game, Alice signals to Cathy that she has the red 3 as either a *Prompt* or a *Finesse*.
- In this situation, it would be foolish for Cathy to play the hypothetical [red 3, red 5] as a *Prompt*, because that is a card that has been hanging around in her hand for the whole game.
- In practice, Cathy would always play *Finesse Position* instead of *Prompting*, partly because of *Context*, but also explicitly because of NPA.
- Stated in terms of expected value:
  - 95% of *Bluffs* are normal *Bluffs* --> Players can *Finesse* red 3 without causing a *Wrong Prompt*.
  - 5% of *Bluffs* are *Known Bluffs* --> Players cannot *Finesse* and must give a less efficient clue.
- If the percentages were closer to each other, then it might make sense to for players to always account for *Known Bluffs*. But in reality, we are handicapping ourselves in the general case in order to try and accommodate the *Known Bluff* case. So let's not do that.

<br />

## Interaction with *OOO* (advantage)

As further evidence for preferring a normal *Bluff* over a *Known Bluff*, consider the interaction with OOO.

- First, see the section for the [Out-of-Order Play Clue](level_19.md##the-out-of-order-play-clue-triple-o--ooo)
- For example, in a 3-player game:
  - Cathy's hand is as follows, from left-to-right: `red 4, red 3, blue 1, green 3, red 1`
  - Green 3 is already played on the stacks.
  - In this hand configuration, cluing number 3 would be a *Bad Touch* on the green 3.
  - In this hand configuration, cluing number 4 would cause the red 4 to be misplayed as a green 4.
  - In this hand configuration, cluing red would cause the focus of the clue to incorrectly be on the red 4 instead of the red 3.
  - Alice decides to clue red to Cathy.
  - Bob can see that Cathy **did not** have *Free Choice* and that the red 3 is blocked by all possible clues.
  - Thus, Bob can reason that Alice is intending for Bob to clue number 4 as a *OOO Play Clue*. It would be a mistake for Bob to blind-play his *Finesse Position* as a *Known Bluff*.

When Cathy imagines whether or not to entertain NPA, she knows that OOO situations like this example will happen in a non-trivial amount of card configurations.  This means that many of Cathy's card configurations will completely prevent a *Known Bluff* with *Free Choice* from ever being triggered. This interaction does not *prevent* them from occurring, but it does reduce them. It's another good reason to disregard *Known Bluffs*.

<br />

## Interaction With Trash (advantage or disadvantage, depending on the situation)

- NPA can also cause players to mark *Trash* in their hand.
- For example, in a 3-player game:
  - Red 3 is played on the stacks. No other cards are played on the stacks.
  - Cathy's hand is as follows, from left-to-right: `red 1, blue 1, blue 1, green 1, red 5`
  - Alice clues red to Cathy, touching the red 1 on slot 1 and the red 5 on slot 5. The clue is *Chop-Focused* on the red 5 on slot 5.
   - Bob sees that this is a weird clue, because Alice has a *Free Choice* between cluing red and cluing number 5. Bob sees that Alice should have clued number 5. Instead, Alice is intentionally focusing the wrong card.
  - Bob blind-plays his *Finesse Position* as a *Finesse* on the red 4. Instead, it is a blue 1 and it successfully plays on the stacks.
  - Cathy marks her slot 5 card as the red 5. (Cathy knows that Alice performed a *Bluff*. Since slot 5 was the focus of the clue, it must be the *one-away-from-playable* red card.)
  - Cathy knows that from the *No Playables Assumption*, Cathy's slot 1 card is not allowed to be the red 4.
  - Thus, since the combination of *Good Touch Principle* and NPA remove all of the possibilities, Cathy knows that the card must be trash. Cathy can safely discard the red 1 from slot 1 (instead of discarding her chop card).
- Because of how NPA can cause players to mark cards are trash, doing it to real playable cards would be illegal and really bad. (This is the disadvantage of having the convention "turned on".)
- For example, in a 3-player game:
  - The setup is the same as in the previous example, except Cathy has a red 4 (the next playable red card) instead of red 1 (trash).
  - Bob sees that Alice has *Free Choice* and that Alice should be cluing number 4 to get the red 4. Thus, Bob knows that this is a *Duplication Finesse* (i.e. he is playing the other copy of the red 4). 
  - Bob blind-plays his *Finesse Position* expecting to play the red 4. Instead, it is a blue 1 and it successfully plays on the stacks.
  - Just like in the previous example, Cathy marks her slot 1 card as a trash card, and then she discards the red 4. Oops!
  - This was Alice's fault for giving the clue, not Cathy's fault.
- When evaluating the pros and cons of this interaction, you should be swayed by the fact that a 2-for-1 clue that promises a *Trash* card is "good enough", as opposed to a move that promises a 3-for-1. (Optimizing for 3-for-1s is unnecessary, especially for *No Variant* games. The goal here should be to enforce behavior that goes towards maximizing average score in "No Variant" and "Rainbow (6 Suits)".)

**Comment by pianoblook** [comment]
I can appreciate the value + advantages of assuming r5 over r3 in the first set of examples. 

That said, it appears to be a large and unsound logical leap to have this same assumption extend to assuming trash instead of r4 in the last example ("interaction with trash").

- You say that  Cathy should disregard the card ever being r3 because "_Known Bluffs_ using Free Choice are extremely rare". Okay, sure - so what should she assume about the other card? 
- It seems like you believe she should still assume that _Good Touch_ is not being violated, and therefore mark it as r5. Okay, 
- But then in the "Interaction with Trash" section you suddenly claim that "_The Never Known Bluff Assumption_ **can also cause players to mark Trash in their hand**."

But why? That claim - that she should assume trash - just came out of nowhere. If she's promised good touch (notes for exactly r5) in the first set of examples, why not be promised good touch (notes for exactly r4) in the second?

**Comment by Zamiell** [comment]
> You say that Cathy should disregard the card ever being r3 because "Known Bluffs using Free Choice are extremely rare".

do you mean r4?

**Comment by pianoblook** [comment]
> > You say that Cathy should disregard the card ever being r3 because "Known Bluffs using Free Choice are extremely rare".
> 
> do you mean r4?

Nope, r3. From the last bullet from your first example:
> What note should Cathy write on her slot 2 card? Cathy starts to write a note of [red 5], but then she stops and considers an alternate hypothetical.

If not for this N.K.B.Assumption, she would write [r3, r5] notes

**Comment by pianoblook** [comment]
The full sentence I was quoting: 
> However, Cathy also knows that **Known Bluffs using Free Choice are extremely rare**. As an unstated convention that is not officially in the document (but learned through experience in playing with the group), Cathy knows that **she should disregard this rare possibility, and instead just simply write a note of [red 5].**

So why is she assuming Good Touch and writing [red 5]? Why not discard it as Known Trash, like she appears to be able to do in the other example?

**Comment by 0nen** [comment]
> If Alice really did perform a Known Bluff with Free Choice, then she Lied to Cathy about the identity of the slot 2 card. In that case, the team must give a follow-up Fix Clue to Cathy at some point in the future.

No, a followup clue is not strictly necessary. When r5 appears on the board in any player's hand besides Cathy, then no clue needed. The situation resolves itself and is extremely clue efficient because Cathy will know which one is r3 and which is r4 with no additional clues given, and it can even tell other players that they drew r5 (or already had it in their hand if Cathy immediately plays r3 after the blind play).

In an alternate scenario, if Cathy draws any 5, a 5 clue to Cathy resolves the situation and is not a fix clue (a 3 or 4 clue). After receiving the 5 clue, Cathy can play r3 and r4, because Cathy knows which is r4 already. This isn't a fix, it's the natural way the game will play out. Again, extremely clue efficient.

**Comment by Zamiell** [comment]
> But why? That claim - that she should assume trash - just came out of nowhere. If she's promised good touch (notes for exactly r5) in the first set of examples, why not be promised good touch (notes for exactly r4) in the second?

why? because NKBA is more important than GTP. in other words, after a bluff occurs, I am proposing that players should use the following algorithm:

1) list all physical possibilities (pure empathy)
2) subtract NKBA cards (i.e. the N+1 card of the suit, where N is the current played card on the stacks)
3) subtract cards that would not apply due to Good Touch Principle

following this algorithm for example 1, we get:

1) [red 1, red 2, red 3, red 4, red 5]
2) [red 1, red 2, red 4, red 5] (because red 2 is played on the stacks)
3) [red 5]

following this algorithm for the final example, we get:

1) [red 1, red 2, red 3, red 4, red 5]
2) [red 1, red 2, red 3, red 5] (because red 3 is played on the stacks)
3) [empty set]

the empty set is equal to trash

**Comment by jakestiles** [comment]
I'm not yet expressing an opinion on this NKBA proposal because I need more info. See below for some info I need:

In my opinion, the example below is legal now and would be legal even if this current proposal is accepted as is. Does anyone agree/disagree?

- Red 3 is on the stacks in a 3-player game. 
- Alice clues red to Cathy touching red 5 on chop (focus) and a non-critical red 4 on slot 2.
- Bob sees that Alice could have clued 4 or 5 allowing Bob to clue red, and there would never have been a fix required. Bob debates whether he is required to do a fix or if he can just try to play the other copy of red 4 from his finesse position. Since there is nothing in the conventions that compels Bob to fix, and since playing finesse position is a line not requiring a fix clue, and since it is logical that Alice wants red 5 to play faster for tempo reasons, Bob plays finesse position.
- Bob is surprised when it turns out that he played green 1 successfully to the stack.
- Cathy writes red 5 on her chop card knowing this was a bluff. She then thinks about what to write on her slot 2 card. She writes red 4 due to good touch principle and immediately plays it.

Cathy never entertains the possibility of a Known Bluff and thus follows Zamiel's new Never Known Bluff Assumption perfectly.

**Comment by Zamiell** [comment]
> No, a followup clue is not strictly necessary. When r5 appears on the board in any player's hand besides Cathy, then no clue needed. The situation resolves itself and is extremely clue efficient because Cathy will know which one is r3 and which is r4 with no additional clues given, and it can even tell other players that they drew r5 (or already had it in their hand if Cathy immediately plays r3 after the blind play).
> In an alternate scenario, if Cathy draws any 5, a 5 clue to Cathy resolves the situation and is not a fix clue (a 3 or 4 clue). After receiving the 5 clue, Cathy can play r3 and r4, because Cathy knows which is r4 already. This isn't a fix, it's the natural way the game will play out. Again, extremely clue efficient.

this is a great point, but it sounds like you might misunderstand how *Good Lie Principle* works.
in our group, *Good Lie Principle* is one of the *First Principles* - the most important and foundational way that we play.
in Hanabi, you can sometimes perform extended *Lies* to your teammates, and things will end up working out great in the end.
the example that you just typed out is perfect example of this!
after performing an extended *Lie*, and having everything work out, it can be alluring to think that they *always* work out, or that they *always* are a good idea.
as you say, they are "extremely clue efficient!"
but our conventions are not optimized for efficiency - that is an anti goal.
our conventions are actually optimized for players having correct and accurate information about the cards that they have in their hand, and the general game state.
this allows expert players to safely navigate complicated situations, without having the complexity of having to entertain alternate possibilities, always allowing them to plan out really solid *Lines* into the future that will **reliably** work out.
in our group, the goal is to work as a team, play predictably, tell the truth, and win 100% of all the games we play.
lying is antithetical to that - the "extreme clue efficiency" is just simply not needed and causes a **reduction** of perfect scores, not an increase of win rate.

so yes - you are correct in that Alice *could* perform a lie to Cathy about the identity of Cathy's slot 2 card, and if she did, she wouldn't necessarily *have* to give a *Fix Clue* later on.
but that is a distraction from the main point at hand - Alice should never do this unless things are going very poorly and Alice is desperate to win the game, so it isn't relevant to this discussion

**Comment by Zamiell** [comment]
> Cathy never entertains the possibility of a Known Bluff and thus follows Zamiel's new Never Known Bluff Assumption perfectly.

did you read my entire post? i already typed up that exact hypothetical word for word (almost) already above, correct me if i am wrong

**Comment by jakestiles** [comment]
> > Cathy never entertains the possibility of a Known Bluff and thus follows Zamiel's new Never Known Bluff Assumption perfectly.
> 
> did you read my post? i already typed up that exact hypothetical word for word, correct me if i am wrong

My example is different. As it doesn't require a known bluff interpretation for Bob to blind play. It could just be a "Alice wants my r4 and Cathy's r5 for tempo reasons" interpretation.

**Comment by Zamiell** [comment]
> It could just be a "Alice wants my r4 and Cathy's r5 for tempo reasons" interpretation.

We call that an *End-Game Distribution Finesse*. In my final example, Bob blind-plays blue 1 as either a *Known Bluff* with *Free Choice* or an *End-Game Distribution Finesse*, although the second part is omitted for brevity. Thus, it is identical to your hypothetical.

**Comment by timotree3** [comment]
> so yes - you are correct in that Alice could perform the Known Bluff and Lie to Cathy about the identity of her slot 2 card, and if Alice did that, she wouldn't necessarily have to give a Fix Clue later on.
but that is a distraction from the main point at hand - Alice should never do this unless things are going very poorly and Alice is desperate to win the game, so it isn't relevant to this discussion

I think it's very relevant. The question at hand for me is how the advantage of fewer superpositons and no wrong prompts compares to the disadvantage of not being allowed to give the known bluff. @0nen is pointing out that without this convention there are already no wrong prompts, at least in the case where the 2 of that suit is already on the stacks. If somebody clues r5 then the card is already known r3

**Comment by jakestiles** [comment]
I never said my hypothetical was in the end game.

Please don't omit anything for brevity. Don't you now you need to fix it and change the name of this proposal to Never Known Bluff & Also Never Distribution Finesse Assumption?

**Comment by Zamiell** [comment]
> Please don't omit anything for brevity. 

i updated the final example to clarify.

> Don't you now you need to change the name of this proposal to Never Known Bluff & Also Never Distribution Finesse Assumption?

no. because this convention is only about cathy, it is not about bob.
i believe i put that in bold above, its very important that you remember this

**Comment by Zamiell** [comment]
> I think it's very relevant. The question at hand for me is how the advantage of fewer superpositons and no wrong prompts compares to the disadvantage of not being allowed to give the known bluff. @0nen is pointing out that without this convention there are already no wrong prompts, at least in the case where the 2 of that suit is already on the stacks. If somebody clues r5 then the card is already known r3

once the entire suit becomes visible towards the end of the game, then no wrong prompts would obviously happen.
but this isn't the default case when one wants to give a finesse

**Comment by jakestiles** [comment]
> ## The Never Known Bluff Assumption Interaction With Trash (advantage or disadvantage, depending on the situation)
> * _The Never Known Bluff Assumption_ can also cause players to mark _Trash_ in their hand.
> * For example, in a 3-player game:
>   
>   * Red 3 is played on the stacks. No other cards are played on the stacks.
>   * Cathy's hand is as follows, from left-to-right: `red 1, blue 1, blue 1, green 1, red 5`
>   * Alice clues red to Cathy, touching the red 1 on slot 1 and the red 5 on slot 5. The clue is _Chop-Focused_ on the red 5 on slot 5.
>   * Bob sees that this is a weird clue, because Alice has a _Free Choice_ between cluing red and cluing number 5. Bob sees that Alice should have clued number 5 **as a save clue.** Instead, Alice is intentionally **giving a play clue and touching a trash card**.
>   * Bob blind-plays his _Finesse Position_ as a _Finesse_ on the red 4. Instead, it is a blue 1 and it successfully plays on the stacks.
>   * Cathy marks her slot 5 card as the red 5. (Cathy knows that Alice performed a _Bluff_. Since slot 5 was the focus of the clue, it must be the _one-away-from-playable_ red card.)
>   * Cathy knows that from the _Never Known Bluff Assumption_, Cathy's slot 1 card is not allowed to be the red 4.

How does NKBA tell Cathy slot 1 can't be red 4?

NKBA says: "By default, the players writing the notes should not assume that the Bluff was a Known Bluff initiated with Free Choice, since that move is extremely rare."

Cathy doesn't have to assume a known bluff for red 4 to be on slot 1! She can hypothesize that this was a simple (non-Known) bluff.

**Comment by Zamiell** [comment]
oh, i think you are right.
then the convention does need to say that it is the *Never Known Bluff or Distribution Finesse Assumption".
how about we rename it to "No Playables Assumption", NPA.
NPA is the logic that cathy uses after a bluff to construct an algorithm for which to assign card notes.

since distribution finesses are extremely rare (at least as rare as *Known Bluffs* with *Free Choice*, if not more rare), all of the same logic applies from my previous post applies, i.e. defining moves as 3-for-1 is unnecessary

**Comment by timotree3** [comment]
This convention is not good when it just changes the note from `[3, 5]` to `[5]` as pointed out by @0nen and when it changes the note from `[4]` to `[kt]` that seems very counter-intuitive given that good touch principle is so core and usually takes precedence over other principles. It seems marginally good to change a note from `[1, 2, 3, 4, 5]` to `[2, 3, 4, 5]`, but not unclear whether it's worth the marginal decrease in flexibility given to Alice.

Overall it seems like in order not to be net negative this proposal would have to special cased the first two cases I just mentioned, at which point it seems like a lot of complexity for an unclear benefit/possible detriment.

**Comment by jakestiles** [comment]
Thanks for the edits Zam, I think that puts us on the same page with regards to understanding the proposal and its impact. 

I tentatively agree with timotree's most recent comment and am against the NPA proposal. But I will think more on it and try to read more of the incoming discussion.

**Comment by 0nen** [comment]
> this is a great point, but it sounds like you might misunderstand how _Good Lie Principle_ works.

I don't see a Good Lie Principle in the docs when I search. I do see the _Lie Principle_ which states:

> Normally, players try to make their clues as clear as possible and convey the exact identity of a card.
> Rarely, players can use clues to trick other players into thinking that they have cards that they really don't.
> If this untruth resolves immediately (e.g. the very next turn), it is called a Fib. For intermediate and advanced players, Fibs are legal and expected.
> If this untruth does not resolve immediately, it is called a Lie. Lies are illegal. Players should never assume that they are Lied to.

At no point in what I wrote did I suggest anything that would create a situation that does not immediately resolve.

- r4 is the focus of a play clue.
- a blind play occurs
- the user knows the focused card is r4, not r3

*Nothing else is promised, nothing else is lied about. There was no promise about slot 2!* The focus was r4, and it clearly is r4 after the blind play happens.

> after performing an extended _Lie_, and having everything work out, it can be alluring to think that they _always_ work out, or that they _always_ are a good idea.

This is *not* an extended lie. The lie resolves immediately. We go from "This card is playable" -> "This card is one away from playable." The lie is resolved.

So another card was picked up - but no promise was made in relation to that card! In fact it complicates things to make assumptions about what that other card is. With r3 having two copies in any given deck, and r5 having only one copy, 2/3 times that r4 is focused with an extra card picked up (with _Good Touch_), that extra card will in fact be r3, not r5.

It is not hard for players to think that a card can be more than one thing. Just like when we clue a players hand with "red" and touch 3 cards, we only promise what the focused card is. The other two are unknown red cards until later. We are big boys and can handle not knowing what a card is between two possible choices, especially when there has been *absolutely no lie* about what that other card is!

> our conventions are actually optimized for players having correct and accurate information about the cards that they have in their hand, and the general game state.

We get correct and accurate information by understanding and following the strategies as they are documented here. It is ultimately your choice what makes it into the conventions. We agree to play by the same rules because we are following them as they are documented.

> lying is antithetical to that - the "extreme clue efficiency" is just simply not needed and causes a **reduction** of perfect scores, not an increase as you seem to be implying.

Again, it's not a lie. You have it in your mind that it was lied about. The second card is not the focus of the clue. It's not a lie. Read that again. _It's not a lie._

> so yes - you are correct in that Alice _could_ perform the _Known Bluff_ and _Lie_ to Cathy about the identity of her slot 2 card, and if Alice did that, she wouldn't necessarily _have_ to give a _Fix Clue_ later on.
> but that is a distraction from the main point at hand - Alice should never do this unless things are going very poorly and Alice is desperate to win the game, so it isn't relevant to this discussion

In many hard variants you need to be clue efficient to win, and in games you will frequently find yourself at 0 clues wishing you had more. What I am suggesting is not an extended lie, and I don't know how to communicate that to you any clearer than I already have repeatedly.

In my view, it is *simpler* to just know that something is r4 and make no assumptions about the slot 2 card. Telling people what to assume about the other card is more complicated.

**Comment by Zamiell** [comment]
> I don't see a Good Lie Principle in the docs when I search

it got renamed to lie principle so that was a typo

> No, a followup clue is not strictly necessary. 

are you talking about in the case where my convention is turned on, or in the case where my convention is turned off? it isn't clear

**Comment by dobi815** [comment]
Not your term Zamiel, but a way I view this assumption is the "Strong Truth Principle". It would state that from Cathy's POV If Bob's behavior (say a blind play) suggests that Alice's clue could not have been fixed to be the truth (2-for-2, OOO etc.), then it must be the case. 

It results in Cathy concluding that she has No Playables in the hypothetical above. 

I am not a fan of the advantages listed in your post to bring up this principle - for a variety of reasons. 

Complexity [For Cathy]: 
1. Players reason more deeply about the focus of the clue than the rest of their hand. This post argues that they should apply an additional principle (in addition to Good touch) to the rest of their hands

Complexity [For Bob]:
1. The definition of what is a fix to get to the truth varies a lot more and is contextual. Bob needs to evaluate if the intent that Alice has is a fix or a blind play and we do not have _tools to make that decision_ in the convention. It relies on context (this hypothetical above refers to free choice, but often blocking cards makes things murkier)

Complexity [For Alice, Cathy]:
1. If Cathy cannot correctly reason on whether the clue could have been fixed or not (as is often the case), then its on Alice to identify that and conclude that the clue is illegal.

Limited advantage:
1. Some of the advantages come from limiting the superposition (from [3,5] to [5] in the above example). This extra refinement is hardly necessary to win games. Players are currently comfortable with far more complicated superpositions and have the tools to handle them.

Advantages of not having this:
1. A lot of simple bluffs can be initiated with an OOO clue. The common one that i've encountered is a 3 and 2 or 4 and 3 clued OOO to get a bluff (or even a double bluff with 4 and 2) and then go on to play successfully. 
2. The focus card is never misconstrued as playable. Very rarely do players reason or have to play the non focused card. Usually a clue like this results in a no-op.

In game example of a complex scenario that comes from assuming the truth in an ambiguous situation: https://hanab.live/replay/478480#4

**Comment by Zamiell** [comment]
i dont understand what the replay is showing

**Comment by dobi815** [comment]
The replay is showing that Alice gave a clue to Donald and Bob assumed the truth could be made to happen with a fix and a bad thing happened in that game.

**Comment by sjdrodge** [comment]
> > It could just be a "Alice wants my r4 and Cathy's r5 for tempo reasons" interpretation.
> 
> We call that an _End-Game Distribution Finesse_. In my final example, Bob blind-plays blue 1 as either a _Known Bluff_ with _Free Choice_ or an _End-Game Distribution Finesse_, although the second part is omitted for brevity. Thus, it is identical to your hypothetical.

Just want to fix this misconception. The example of finessing Bob's r4 using Cathy's r5 and happening to touch Cathy's r4 at the same time is not a Distribution Finesse which is the following:
```
The Distribution Finesse
On the final round of the game, it is possible to give a Play Clue to a player that would be useless - they will not get a chance to play the card before the game ends.
If a player does this anyway, then it must be communicating something extra. In this situation, it is to be interpreted as a Finesse instead of a Play Clue for the purposes of better satisfying Team Distribution Principle.
```
As we can see, this is unrelated. The situation we're talking about just occurs any time the remaining set of red cards is sufficiently small that Bob knows Alice intends for him to play r4 and Cathy to therefore know the 6th red card is trash. Furthermore, the assertions that this comes up "extremely rarely" and is also "not how we have historically played" are both false, so I'd encourage you to stop making those assertions, Zamiel.

**Comment by Zamiell** [comment]
> The replay is showing that Alice gave a clue to Donald and Bob assumed the truth could be made to happen with a fix and a bad thing happened in that game.

i dont see how that is relevant to what i am proposing

**Comment by Zamiell** [comment]
> As we can see, this is unrelated. The situation we're talking about just occurs any time the remaining set of red cards is sufficiently small that Bob knows Alice intends for him to play r4 and Cathy to therefore know the 6th red card is trash.

ok, then it isn't a Distribution Finesse, but it is instead a Finesse that Satisfies Team Distribution Principle.
call that a TeamDistribution Finesse instead then, if you like.

**Comment by sjdrodge** [comment]
> > As we can see, this is unrelated. The situation we're talking about just occurs any time the remaining set of red cards is sufficiently small that Bob knows Alice intends for him to play r4 and Cathy to therefore know the 6th red card is trash.
> 
> ok, then it isn't a Distribution Finesse, but it is instead a Finesse that Satisfies Team Distribution Principle.
> call that a TeamDistribution Finesse instead then, if you like.

I think this name is poor. Often Alice's motivation has nothing to do with distributing cards and instead she is just giving the simplest clue to get the rest of the red cards. The alternative is using 2 clues (as a single clue to Cathy's hand cannot get both red cards). It's just a normal finesse that works because Cathy will never think the extra red card is good touched, due to all red cards being already accounted for. It's very similar to cluing Cathy 1's touching 3 1's when 4 out of 5 1's have been played. Cathy will not think the 2nd and 3rd 1 are good, so there's no problem.

The part that is controversial is when Alice tries to bluff Bob using this logic, since there seems to be some debate about whether Cathy ought to simply follow Good Touch Principle in that circumstance, or assume instead that Alice would never give this kind of clue.

**Comment by Zamiell** [comment]
call it a *Duplication Finesse* then

**Comment by sjdrodge** [comment]
Regarding the merits of the NPA convention, I will point out that in the Known Bluff example given in the proposal, even without NPA, Cathy can often tell whether it's possible for the touched card to be r3 or not. This is because an ambiguity between a prompt/finesse would require that Cathy has already gone on to discard. Often this means that Cathy will be able to know whether a potential 3's clue to her hand was blocked at the time the bluff was given. If a 3's clue was known to be blocked, then Cathy can exclude r3 anyway and play finesse instead of prompting.

Regarding the important bullet point number 6. `Note that the No Playables Assumption does not prescribe behavior for Bob - it only prescribes behavior for Cathy. Bob should always blind-play his Finesse Position card as a Finesse or a Known Bluff if he sees that Alice clearly does not intend for him to give a OOO Fix Clue.` Let's take a moment to clear up a misconception. It is not possible in general, and certainly not possible in this particular case, to completely separate Bob's prescribed behavior from Cathy's. In a Known Bluff situation, if the prescribed behavior for Cathy is that she should not allow r3 as a possibility for her extra touched card, Bob cannot simply ignore this and knowingly cause a problem by blind-playing. What if r5 is visible in Alice's hand? What if it's visible in Donald's? Should Bob still blind-play knowing that Cathy will treat the r3 as trash? Prescribing behavior for Cathy necessarily also prescribes behavior for Bob and for Alice.

**Comment by Zamiell** [comment]
> Often this means that Cathy will be able to know whether a potential 3's clue to her hand was blocked at the time the bluff was given. 

I don't know about "often". Maybe "sometimes".

Also, the algorithm you just described sounds very complicated and prone to error (i.e. retroactive reasoning), which we've both seen over and over lead to misplays. Maybe another way to phrase this is that I prefer explicitly limiting superpositions over implicitly limiting superpositions.

> Bob cannot simply ignore this and knowingly cause a problem by blind-playing.

If Bob can use context to figure out what Alice really wants, then Bob should do that, whether it is blind-playing a card, giving a clue, or doing something else. If Bob does not have any specific context for the situation (other than that he should probably blind-play his *Finesse Position* card), then Bob should blind-play his *Finesse Position* card and "simply ignore" the complications. This is reinforced by *Guide Principle*, which is already a well-established norm amongst the group that tells Bob to trust Alice by default.

I think this is a great summary of how this convention proposal interacts with Bob, if at all. Now, in order to limit the scope of the discussion, I feel like it is more productive to focus on Cathy.

**Comment by Lel0uch-H** [comment]
Here is an example in favor of the "No playables assumption":

https://hanab.live/replay/478709#23

When indego plays y1 from a brown clue focused on n4, the card can be promised n4 and all other cards can be marked as trash.

<img width="1115" alt="Screenshot 2021-03-24 at 09 38 04" src="https://user-images.githubusercontent.com/81278716/112253468-c85c5c80-8c84-11eb-8efa-9bc2f8cec825.png">

**Comment by Zamiell** [comment]
i'll be the first to say that this probably isn't the best example, since regardless of whether or not this issue is accepted, sucubis should never play anything, due to the nature of how brown works. but sure, with NPA, it would be a lot more clear what is going on

**Comment by Lel0uch-H** [comment]
Here's another example:

https://hanab.live/replay/457404#17

<img width="1110" alt="Screenshot 2021-03-24 at 09 45 19" src="https://user-images.githubusercontent.com/81278716/112253976-c1821980-8c85-11eb-861d-6b72761e0ea4.png">

After the b2 bluff is given, if dobi assumes this is not a known bluff, the self finesse later would work (in game it did not work since dobi assumed the known bluff interpretation where he could have b1 b2 and Fey still blind played from it):

<img width="1106" alt="Screenshot 2021-03-24 at 09 45 44" src="https://user-images.githubusercontent.com/81278716/112254075-f5f5d580-8c85-11eb-8e84-979cd89350f7.png">

**Comment by Zamiell** [comment]
now that is a fantastic example

**Comment by sjdrodge** [comment]
> I think this is a great summary of how this convention proposal interacts with Bob, if at all. Now, in order to limit the scope of the discussion, I feel like it is more productive to focus on Cathy.

Okay, no. We're not going to be ignoring the implications for Bob's and Alice's behavior that flow directly from the prescriptions for Cathy's behavior. It's important to evaluate the entirety of a convention, not just the part we like.

Regarding Lel0uch's example, the slot 3 card is already guaranteed not to be m2 or b1 due to the LHS from earlier in the game, so if b1 had been in slot 1 at the time the blue clue was given, it would just a focus inverted clue. So this is a great example of how context often makes things clear even without an explicit ban on the extra touched card being b1.

That said, it we imagine there's no such context, then I agree it would be extremely strange for slot 1 to be b1, but I don't think it would be weird at all for it to be m2, and I would expect prompts later for m2 to work just fine.

**Comment by Lel0uch-H** [comment]
The interaction of focus inversion and bluffs is out of scope of this discussion. In your hypothetical on turn 17, if dobi has m2 on slot 1 and b2 on slot 3, and the blue clue is given, I can assure you that most players in Fey's seat would assume it is a finesse on b1 rather than a contextual focus inversion play clue on m2.

**Comment by sjdrodge** [comment]
> The interaction of focus inversion and bluffs is out of scope of this discussion. In your hypothetical on turn 17, if dobi has m2 on slot 1 and b2 on slot 3, and the blue clue is given, I can assure you that most players in Fey's seat would assume it is a finesse on b1 rather than a contextual focus inversion play clue on m2.

I'm sure you're right that many players would miss the focus inversion.

However, you're definitely not right that context considerations are outside the scope of the discussion. The fact that context often allows players to eliminate card possibilities affects our evaluation of the pros/cons of the proposal. This example game has multiple turns of context that make b1 impossible (the most obvious being the LHS turn), so it falls into the large category of games that would be unaffected regardless of whether the proposal is accepted or not.

**Comment by Lel0uch-H** [comment]
It sounds like I am repeating myself but I will probably elaborate further:

The LHS on turn 8 only makes the slot 3 (b2) not playable on turn 17, not the slot 1 card (m4). Regardless of the fact that b2 is known unplayable, any clue that touches b2 and m4 is currently being treated by most of the group as being focused on b2 and not m4. So hypothetically regardless of what card you replace m4 with (example m2), it will be treated as a play clue on b2 and hence Fey must respond to the finesse/bluff.

**Comment by sjdrodge** [comment]
> It sounds like I am repeating myself but let me elaborate further:
> 
> The LHS on turn 8 only makes the slot 3 (b2) not playable on turn 17, not the slot 1 card (m4). Regardless of the fact that b2 is known unplayable, any clue that touches b2 and m4 is currently being treated by most of the group as being focused on b2 and not m4. So hypothetically regardless of what card you replace m4 with (example m2), it will be treated as a play clue on b2 and hence Fey must respond to the finesse/bluff.

If dobi can know that the usual focus of the clue is not playable and can therefore successfully focus invert, then it's just a focus inversion play clue. So if m4 were b1, Fey would not play anything. Therefore, when Fey does play, the m4 cannot be b1. The same logic applies to m2, though I agree players are far more likely to miss that (for purely psychological reasons).

**Comment by jack67889** [comment]
I have a few questions about the implications of the NPA.

I assume the NPA only applies to clues that are potentially OOO like (I don’t know a better wording for now), is that correct? For example, nothing is played on the stack Cathy receives a blue clue that touches slot 1 and slot 3, Bob blind-plays his finesse and it’s a bluff, if slot 1 is b2 or b3 could slot 3 still be m1 with the NPA (or if slot 1 is m2 or m3 and slot 3 is b1)? (So that the NPA only applies to cards of the same suit as the focus of the clue.) If not, how is this conceptually different from, for example, bluffs with number clues where a card that is not the focus of the clue is currently playable. Or is it really intended that the NPA also applies to number clues? My guess for now would be that it’s not intended that the NPA applies to number clues, if that’s correct I think the current wording is quite ambiguous, because it could easily be interpreted as that the NPA applies to a 3 bluff for example.

  I assume the NPA also applies to a Double Bluff, 5CD, UTD, 4-Charm? If not, how is that conceptually different for Cathy?

**Comment by jakestiles** [comment]
If this NPA suggestion is good (which I disagree that it is), should we also make a separate issue for a No OCM-ing A 4 Assumption? 

Players, when OCM’d, don’t know what card has been OCM’d, and if filled in later with color, they won’t know if it’s a 2/3/4/5.  Adding the assumption that a 4 will never be OCM’d lets players write down fewer possibilities on their OCM’d card (Interaction with superposition - advantage) and if filled in with color later, it will be able to be played at the correct time more often (interaction with prompts - advantage).

This would be, of course, an awful proposal. 

As a general rule, we shouldn’t voluntarily hamstring our ability to do uncommon but powerful moves in favor of more clarity unless the added assumption is necessary for the particular variant (NPA and No OCM 4A aren’t. See Muddy Rainbow assumption and 1’s Assumption for Pink for assumptions that are more necessary because the variant is difficult) and it doesn’t give up too much power (Which NPA and No OCM4A both do), and it doesn’t take up too much headspace (Which NPA and No OCM4A both do).

**Comment by Zamiell** [comment]
> I assume the NPA only applies to clues that are potentially OOO like (I don’t know a better wording for now), is that correct?

no, NPA applies to all bluffs

> if slot 1 is b2 or b3 could slot 3 still be m1 with the NPA

no it can't... that is the whole point of NPA

>   I assume the NPA also applies to a Double Bluff, 5CE, UTD, 4-Charm? If not, how is that conceptually different for Cathy?

type out an alice and bob story

**Comment by Zamiell** [comment]
> As a general rule, we shouldn’t add artificial assumptions and voluntarily hamstring our ability to do uncommon but powerful moves in favor of more clarity 

jake, this is a normative statement that is disconnected from the group's goals.
we already voluntarily hamstring our ability to do powerful moves all the time.
that's why we have outlawed patch finesses, weak for trash finesses, weak prompts, and so forth.
I'll remind you that yes, clarity **is** one of the main goals of the conventions, and this is a theme that we've consistently chased after over the course of the past 4 years.
more clarity is accomplished whenever we delete an old move that isn't used, or we delete an old move that results in too much complexity and isn't working out.
if a move results in more clarity but less efficiency, then that is a tradeoff you have to actually think about and weigh the pros and cons off, just not dismiss offhand because "it restricts us".

**Comment by jack67889** [comment]
> no it can't... that is the whole point of NPA 

Well, since it’s not a known bluff in that case, your proposal doesn’t cover that situation. The motivation for the proposal as currently stated seems to suggest NPA doesn’t apply in that case.

Okay, thus NPA also promises all the 3s touched by a 3 bluff currently unplayable?

> I assume the NPA also applies to a Double Bluff, 5CE, UTD, 4-Charm? If not, how is that conceptually different for Cathy?

> type out an alice and bob story

Red stack is empty, there are no clued cards:
Cathy receives a red clue on chop and slot 1. Bob charms. Cathy writes down r4 on chop and !r1 on slot 1.

b1 is on the stack, there are no clued cards:
Cathy receives a blue clue on slot 2 and 3. Bob ejects. Cathy writes b5 on slot 2 and !b2 on slot 3.

g1 is on the stack, there are no clued cards:
Cathy receives a green clue on slot 1, 2 and 3. Bob discharges. Cathy writes kt on slot 1 and !g2 on slot 2 and 3.

Et cetera

**Comment by Zamiell** [comment]
> My guess for now would be that it’s not intended that the NPA applies to number clues, if that’s correct I think the current wording is quite ambiguous, because it could easily be interpreted as that the NPA applies to a 3 bluff for example.

that seems reasonable, it sounds like i should update NPC to only apply to number clues, but i am interested in hearing more from others

**Comment by Zamiell** [comment]
> Red stack is empty, there are no clued cards:
Cathy receives a red clue on chop and slot 1. Bob charms. Cathy writes down r4 on chop and !r1 on slot 1.

thats correct because otherwise it would be a OOO finesse (meaning bob would clue number 4)

> b1 is on the stack, there are no clued cards:
Cathy receives a blue clue on slot 2 and 3. Bob ejects. Cathy writes b5 on slot 2 and !b2 on slot 3.

that is not correct, because in this case the "2-or-more-blind-plays" rule supersedes NPA.
(again, this is the way that we have always played, and is probably unmentioned in the doc)
in other words, if bob clued 5 to cathy as a OOO finesse, then bob would still have to blind play blue 3 and blue 4, which is 2 cards

> g1 is on the stack, there are no clued cards:
Cathy receives a green clue on slot 1, 2 and 3. Bob discharges. Cathy writes kt on slot 1 and !g2 on slot 2 and 3.

NPA probably applies here, but instead I will just say that it is undefined, use context.
(having something be undefined is perfectly okay)

**Comment by jack67889** [comment]
> no, NPA applies to all bluffs

At the risk of being overabundant I feel like I should clarify my example. Because it still seems weird to me.

Game of rainbow, b1 is on the stack, there are no clued cards:
Alice clues blue to Cathy.
The clue touches slot 1 and 2.
Bob blind-plays r1.
Cathy marks her slot 1 card as b3.
For her slot 2 card I understand that you say NPA would mean that slot 2 is !b2, but also !m1.

Same example, but now:
Bob blind-plays b2 (instead of r1).
Now it’s a finesse.
Would Cathy still write !m1 on her slot 2 or doesn’t NPA apply to finesses?

If it doesn’t apply to finesses:
Should Bob keep bind-playing after r1 if Alice violates NPA? To tell Cathy the other card is actually playable and turn the bluff in a layered finesse.
That would be weird.

The same reasoning applies to 3 bluffs for example.

**Comment by Zamiell** [comment]
> At the risk of being overabundant

i dont mind =p

> Cathy marks her slot 1 card as b3.

um, wouldnt she mark it as [b3, m2, m3]?

**Comment by Zamiell** [comment]
>  It's important to evaluate the entirety of a convention, not just the part we like.

I'm aware of that, but do you agree that what I said sums up the entire interaction with Bob?

**Comment by sjdrodge** [comment]
> > It's important to evaluate the entirety of a convention, not just the part we like.
> 
> I'm aware of that, but do you agree that what I said sums up the entire interaction with Bob?

I do not.

**Comment by Zamiell** [comment]
jack you should reply to my questions instead of just putting an emoji, as i am not notified of the latter

> Would Cathy still write !m1 on her slot 2 or doesn’t NPA apply to finesses?

as written, NPA specifically has to do with *Bluffs*, not *Finesses*
NPA does not apply to *Finesses*

> Should Bob keep bind-playing after r1 if Alice violates NPA? To tell Cathy the other card is actually playable and turn the bluff in a layered finesse.

no.
i am not proposing a new move.
if you want to violate NPA you can, but it doesn't call for any additional action other than committing the team to give a fix clue later on

**Comment by Zamiell** [comment]
> I do not.

what did I miss?
